core/templates/components/skill-selector/skill-selector.component.ts

```diff
@@ -194,9 +195,14 @@ export class SkillSelectorComponent implements OnInit {
   }
 
   searchInUntriagedSkillSummaries(searchText: string): SkillSummary[] {
-    let skills: string[] = this.untriagedSkillSummaries.map(val => {
-      return val.description;
-    });
+    let skills: string[] = this.untriagedSkillSummaries
+      .filter(
+        val => !this.skillIdsToExclude ? true :
```

Comment:
Why do you need the guard clause here -- in which case will skillIdsToExclude not be provided? Can we make sure it's always provided for all usages of the skill-selector component (even if it's an empty dict)? (Besides, you've typed it above with `!` i.e. certifying that it's non-null.)

Reply:
@seanlip Why I did this was because one of the test failed while pushing. I got the error TypeError: Cannot read properties of undefined (reading 'hasOwnProperty') when it tried to search for untriaged skill summaries meaning that skillIdsToExclude is undefined in this test. This is, however, something I don't think can happen since skillIdsToExclude always is initialized even if it is empty before calling SkillSelectorComponent. 

I found the test that fails and saw that untriagedSkillSummaries which also is used in the searchInUntriagedSkillSummaries function is defined before the test is made. Do I need also define skillIdsToExclude here before the test?
 
 Test Code:

```
it('should search in untriaged skill summaries and return' +
    ' filtered skills', () => {
    component.untriagedSkillSummaries = [
      SkillSummary.createFromBackendDict({
        id: '1',
        description: 'This is untriaged skill summary 1',
        language_code: '',
        version: 1,
        misconception_count: 2,
        worked_examples_count: 2,
        skill_model_created_on: 121212,
        skill_model_last_updated: 124444
      }),
      SkillSummary.createFromBackendDict({
        id: '2',
        description: 'This is untriaged skill summary 2',
        language_code: '',
        version: 1,
        misconception_count: 2,
        worked_examples_count: 2,
        skill_model_created_on: 121212,
        skill_model_last_updated: 124444
      })
    ];

    expect(component.searchInUntriagedSkillSummaries(
      'skill summary 2')).toEqual([
      SkillSummary.createFromBackendDict({
        id: '2',
        description: 'This is untriaged skill summary 2',
        language_code: '',
        version: 1,
        misconception_count: 2,
        worked_examples_count: 2,
        skill_model_created_on: 121212,
        skill_model_last_updated: 124444
      })
    ]);
  });
```

Reply:
@Olivestam I would recommend: (a) first, check all callsites of the component and ensure that skillIdsToExclude is guaranteed to be initialized; (b) ensure that skillIdsToExclude is typed as a non-optional parameter; (c) base your test data on what actually is used within the code (i.e. have the actual code drive how you set up the test, not the other way round -- so, if the code is guaranteed to be passed skillIdsToExclude then tests should follow that as well). Does this help?

Reply:
Thank you! I have now ensured that skillIdsToExclude is initialized before all calls to that component and also initialized skillIdsToExclude in the test.

Reply:
Done